stdlib/3.4/asyncio/tasks.pyi

```diff
@@ -19,10 +24,24 @@ def as_completed(fs: Sequence[_FutureT[_T]], *, loop: AbstractEventLoop = ...,
 def ensure_future(coro_or_future: _FutureT[_T],
                   *, loop: AbstractEventLoop = ...) -> Future[_T]: ...
 async = ensure_future
-# TODO: gather() should use variadic type vars instead of _TAny.
-_TAny = Any
-def gather(*coros_or_futures: _FutureT[_TAny],
-           loop: AbstractEventLoop = ..., return_exceptions: bool = False) -> Future[List[_TAny]]: ...
+@overload
+def gather(coro_or_future1: _FutureT[_T1],
+           loop: AbstractEventLoop = ..., return_exceptions: bool = False) -> Future[Tuple[_T1]]: ...
```

Comment:
Seems like it actually always returns a `List`. Unfortunately, we can't specify how many elements a `List` contains in the type system.

Maybe it's acceptable to lie here and pretend that it returns a tuple, but https://docs.python.org/3/library/asyncio-task.html#asyncio.gather does explicitly say that it returns "a list".